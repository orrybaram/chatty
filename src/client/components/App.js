import React, { Component, PropTypes } from 'react';
import { observer } from 'mobx-react';
import ChatWindow from './ChatWindow';
import '../styles/index.css';

const propTypes = {
  stores: PropTypes.object
};

@observer
class App extends Component {
  constructor(props) {
    super(props);

    const { UserStore, MessageStore, ChatWindowStore } = props.stores;

    this.userOne = new UserStore(0);
    this.userTwo = new UserStore(1);
    this.messageStore = new MessageStore(0);
    this.chatOne = new ChatWindowStore(this.userOne, [this.userTwo], this.messageStore);
    this.chatTwo = new ChatWindowStore(this.userTwo, [this.userOne], this.messageStore);
  }
  render() {
    return (
      <div className='app'>
        <ChatWindow
          chatStore={this.chatOne}
        />
        <ChatWindow
          chatStore={this.chatTwo}
        />
      </div>
    );
  }
}

App.propTypes = propTypes;
export default App;
