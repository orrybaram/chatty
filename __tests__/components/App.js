/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../src/client/components/App';

describe('App', () => {
  it('renders correctly', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
