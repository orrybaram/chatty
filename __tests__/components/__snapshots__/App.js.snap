exports[`App renders correctly 1`] = `
<div
  className="app">
  <div>
    Im a chat window
  </div>
  <div>
    Im a chat window
  </div>
</div>
`;
